### PR TITLE
Document new recording rule format

### PR DESCRIPTION
**Pull request by brian-brazil**
I put the alert format spec in here too, as it'd be odd to split it out.

### CODE REVIEW

File: docs/configuration/recording_rules.md [hunk 1]
```diff
@@ -44,22 +45,74 @@ the original expression every time it is needed. This is especially useful for
...
+Recording and alerting rules exist in a rule group. Rules within a group are
+run sequentially at a regular interval.
 
-    <new time series name>[{<label overrides>}] = <expression to record>
+The syntax of a rule file is:
 
-Some examples:
+```yaml
+groups:
+  [ -<rule_group> ]
```
Comment (grobie) [inline docs/configuration/recording_rules.md]: Space after `-`.

File: docs/configuration/recording_rules.md [hunk 2]
```diff
@@ -44,22 +45,74 @@ the original expression every time it is needed. This is especially useful for
...
+  [ - <rule> ... ]
+```
+
+### `<rule>`
 
-    # Saving the per-job HTTP in-progress request count as a new set of time series:
-    job:http_inprogress_requests:sum = sum(http_inprogress_requests) by (job)
+The syntax for recording rules is:
+```
+  # The name of the time series to output to. Must be a valid metric name.
```
Comment (grobie) [inline docs/configuration/recording_rules.md]: We don't indent the syntax for config parts in `configuration.md` so far. Does it make sense to start this now?

File: docs/configuration/recording_rules.md [hunk 3]
```diff
@@ -44,22 +45,74 @@ the original expression every time it is needed. This is especially useful for
...
+
+  # Alerts are considered firing once they have been returned for this long.
+  # Alerts which have not yet fired for long enough are considered pending.
+  [ for: <duration> | default = 0s ]
 
-    # Drop or rewrite labels in the result time series:
-    new_time_series{label_to_change="new_value",label_to_drop=""} = old_time_series
+  # Labels to add or overwrite for each alert.
+  labels:
+    [ <labelname>: <tmpl_string> ]
```
Comment (grobie) [inline docs/configuration/recording_rules.md]: `tmpl_string` needs to be defined somewhere.

File: docs/configuration/recording_rules.md (hunk 1 above)
Comment (brian-brazil) [reply docs/configuration/recording_rules.md]: Fixed

File: docs/configuration/recording_rules.md (hunk 2 above)
Comment (brian-brazil) [reply docs/configuration/recording_rules.md]: Fixed

File: docs/configuration/recording_rules.md (hunk 3 above)
Comment (brian-brazil) [reply docs/configuration/recording_rules.md]: Ah, I was thinking of the AM definition. Added it to the Prometheus config definitions.

File: docs/configuration/recording_rules.md [hunk 4]
```diff
@@ -44,22 +45,74 @@ the original expression every time it is needed. This is especially useful for
...
+```yaml
+groups:
+  - name: example
+    - record: job:http_inprogress_requests:sum
+      expr: sum(http_inprogress_requests) by (job)
+```
+
+### `<rule_group>`
+```
+# The name of the group. Must be unqiue within a file.
```
Comment (roidelapluie) [inline docs/configuration/recording_rules.md]: unique*

File: docs/configuration/recording_rules.md [hunk 5]
```diff
@@ -44,22 +45,74 @@ the original expression every time it is needed. This is especially useful for
...
+name: <string>
+
+# How often rules in the group are evaluated.
+[ interval: <duration> | default = global.evaulation_interval ]
```
Comment (roidelapluie) [inline docs/configuration/recording_rules.md]: evaluation*

File: docs/configuration/recording_rules.md [hunk 6]
```diff
@@ -44,22 +45,74 @@ the original expression every time it is needed. This is especially useful for
...
 
-    # Saving the per-job HTTP in-progress request count as a new set of time series:
-    job:http_inprogress_requests:sum = sum(http_inprogress_requests) by (job)
+The syntax for recording rules is:
+```
+# The name of the time series to output to. Must be a valid metric name.
+record: <string>
+
+# The PromQL expression to evaluate. Every evaluation cycle this is
+# evaluated at the current time, and the result recorded in the time
```
Comment (juliusv) [inline docs/configuration/recording_rules.md]: nomenclature nit: "...recorded as a new set of time series with the metric name as given by 'record'".

**Comment by juliusv** [comment]
:+1: besides comments

**Comment by brian-brazil** [comment]
All fixed, thanks.

**Pull request merged**